WordPress/pattern-directory
# Fix editor and pattern preview styles, use Twenty Twenty-Three for previews

Fixes #554 — This switches the pattern preview code to use Twenty Twenty-Three, so that all available layout and style settings are supported. This also adds the layout styles to the creator view, so that wide & full width elements work. Lastly, this changes how the preview is rendered. Instead of aborting after `template_redirect`, this creates a fork of the `template-canvas.php` and renders a fake template, allowing all relevant filters to run and inject the correct container & layout styles.

### Screenshots

| (trunk) Editor | (trunk) Preview | (pr) Editor | (pr) Preview |
|-----|-----|-----|------|
| ![trunk-site-logo-editor-1280](https://user-images.githubusercontent.com/541093/221309539-1bca3aea-f534-43d3-9d40-1780b43b8944.png) | ![trunk-site-logo-view-1280](https://user-images.githubusercontent.com/541093/221309541-93139aa4-1987-4124-9747-1ce47a0bc3ef.png) | ![branch-site-logo-editor-1280](https://user-images.githubusercontent.com/541093/221309516-6f30c4ef-ce85-4bb4-a209-234c9b8f33cf.png) | ![branch-site-logo-view-1280](https://user-images.githubusercontent.com/541093/221309521-d1115bd6-601f-411b-9bab-8bb191246b52.png) |
| ![trunk-block-gap-editor-1280](https://user-images.githubusercontent.com/541093/221309533-798cd12d-35ef-4b5b-a9d5-fba534a9c33e.png) | ![trunk-block-gap-view-1280](https://user-images.githubusercontent.com/541093/221309536-096d6457-de5a-40d5-8d4f-e59b11497d31.png) | ![branch-block-gap-editor-1280](https://user-images.githubusercontent.com/541093/221309496-bfa877a4-a735-4738-9cfa-b55a06d74a63.png) | ![branch-block-gap-view-1280](https://user-images.githubusercontent.com/541093/221309511-b843ab83-8ace-42be-9396-e1348f05005e.png) |
| ![trunk-three-widths-editor-1280](https://user-images.githubusercontent.com/541093/221309542-33be96d6-6b4e-4119-ac41-7dfa0fc8342e.png) | ![trunk-three-widths-view-1280](https://user-images.githubusercontent.com/541093/221309546-180028bb-2300-4814-b595-0eae27e0d23e.png) | ![branch-three-widths-editor-1280](https://user-images.githubusercontent.com/541093/221309525-c4410591-7793-47af-bcae-8609930d5dcf.png) | ![branch-three-widths-view-1280](https://user-images.githubusercontent.com/541093/221309530-b2e888dc-a13b-4a84-9460-785372179825.png) |

### How to test the changes in this Pull Request:

1. Create patterns with various layout options, specifically using block gap & widths
2. Try buttons, other design changes
3. The creator view and the pattern preview should look the same, no drastic differences
